extensions/chathistory.md

```diff
@@ -36,10 +40,22 @@ Full support for this extension requires support for the [`batch`][batch], [`ser
 
 The `draft/chathistory` capability MUST be negotiated. This allows the server and client to behave differently when history is available and can be queried. In particular, if a client requests this capability, the server (or bouncer) SHOULD NOT play back any history that would otherwise be sent automatically.
 
-An ISUPPORT token MUST be sent to the client to state the maximum number of messages a client can request in a single command, e.g., `CHATHISTORY=50`. If `0`, the client SHOULD assume that there is no maximum number of messages.
-
 The `draft/event-playback` capability MAY be negotiated. This allows the client to signal that it is capable of receiving and correctly processing lines that would normally produce a local state change (such as `JOIN` or `MODE`) in its history batches.
 
+### ISUPPORT tokens
+
+This specification defines two ISUPPORT tokens.
+
+`CHATHISTORY` MUST be sent to the client to state the maximum number of messages a client can request in a single command, e.g., `CHATHISTORY=50`. If `0`, the client SHOULD assume that there is no maximum number of messages.
+
+`MSGREFTYPES` SHOULD be sent, with a comma-separated list of types of message references they support as parameter to `CHATHISTORY` commands, in order of decreasing preference. Clients MUST ignore any type they do not support. The currently defined list of types is:
+
+* `timestamp`, to reference messages by their timestamp
+* `msgid`, to reference them by their `msgid`
+
+If `MSGREFTYPES` is missing, clients SHOULD assume the server supports both and has no preference.
```

Comment:
This "legacy path" isn't quite correct IMHO. If the server doesn't send `msgid` tags, then the client has no way to use the `msgid` reference type. Same goes for the `time` tag. Maybe that should be clarified? ie, state that the server will support `timestamp` refs if `time` tags are sent, and will support `msgid` refs if `msgid` tags are sent?

Reply:
Is it worth mentioning it in the spec, though? If either tag is missing, then clients don't really have a choice anyway

Reply:
Right, just feels a bit weird to specify that the server supports both. I guess it's not that of a big deal since it's a SHOULD.